python/rmmbr/main_test.py

```diff
@@ -18,13 +26,17 @@ async def f(x):
 
         f_cached = cacher(f)
         await f_cached(3)
-        results = await asyncio.gather(*map(f_cached, [3, 3, 2, 1]))
+        results = [
+            await wait_all_tasks() or await f_cached(n) for n in [3, 3, 2, 1]
```

Comment:
`wait_all_tasks` doesn't return anythin so i'm not sure i follow this condition

Reply:
It's how I inline running both functions but getting the second one's result, would take suggestions here

Reply:
why not `await wait_all_tasks` in the line before?

Reply:
You have to do that before or after every call